src/KOKKOS/kokkos_type.h

```diff
@@ -769,6 +769,38 @@ typedef tdual_virial_array::t_dev_um t_virial_array_um;
 typedef tdual_virial_array::t_dev_const_um t_virial_array_const_um;
 typedef tdual_virial_array::t_dev_const_randomread t_virial_array_randomread;
 
+// Spin Types
+
+//3d SP_FLOAT array n*4
+#ifdef LMP_KOKKOS_NO_LEGACY
+typedef Kokkos::DualView<X_FLOAT*[4], Kokkos::LayoutLeft, LMPDeviceType> tdual_sp_array;
```

Comment:
@julient31 can you make this a general type `tdual_float_1d_4`? Then in the Spin style you can typedef it to a new name.